doc/json-schema.rst

```diff
@@ -0,0 +1,1059 @@
+################################
+XGBoost JSON Schema, Version 1.0
```

Comment:
What is the plan to maintain this? I can see this going out of date quickly without some kind of CI enforcement. This looks like we are exposing the complete serialisation format such that someone could construct a working xgboost model completely external to xgboost. This could be a very useful feature, but also with some cost to support. Note that up until now we have not provided a schema for our binary format, only tried to enforce backwards compatibility.

Reply:
@RAMitchell We can automate the schema enforcement if we express our model schema using JSON Schema. There are tools available to validate JSON files against a given JSON schema.